### DISCUSSION TITLE
Prompt Weighting

### DISCUSSION THREAD

**Discussion by Muskelmagier** [hlky/stable-diffusion-webui#525] [Q&A]
In the Features list Prompt Weighting is mentioned, but there is no description in how to use it

**Comment by James009D** [comment]
So, I could be wrong here but it seems to work when you use something like cyberpunk:0.25 and then Cyberpunk would have a 25% weight in the image. I don't know exactly how the math adds up or how it works with the rest of your prompt words; I assume normal words have a 100% weight.

I guess we could run some quick experiments... lets see!

**Reply by James009D** [reply]
So, I started with 3 variations of the Cyberpunk weight and got some interesting results.

Cyberpunk:0.0 just broke the image and made some ugly faces; it broke the entire prompt.
Cyberpunk:0.5 also broke the image and made some ugly faces; it broke the entire prompt.
Cyberpunk:0.9 had her wearing a black tank top which wasn't super cyberpunk but it's black.
![00001-100_k_lms_9001_0 00](https://user-images.githubusercontent.com/99635973/188306567-27d723b2-4713-49ae-8d6b-6d7c48d0ffaa.png)
Cyberpunk:2.0 had her wearing a green uniform
![00001-90_k_lms_9001_0 00](https://user-images.githubusercontent.com/99635973/188306583-536d7397-e65e-4b8d-a584-9abba1b18d96.png)
Cyberpunk:2.0 had her wearing a more futuristic black uniform but I think it removed details from the rest of the photo
![00003-108_k_lms_9001_0 00](https://user-images.githubusercontent.com/99635973/188306632-0f888946-8468-4d87-8271-56fe8706ebb3.png)
Cyberpunk:5.0 had her wearing a had it adding some details to it but meh.
![00000-121_k_lms_9001_0 00](https://user-images.githubusercontent.com/99635973/188306687-4db8f7d7-5a1a-4eb9-9d83-6e2762a2d1f1.png)

I think you are supposed to use weights with other weights and have it pick and fight between them but I have not mastered it yet. I'd like to note that I used VERY low resolution images without many steps to test these.

**Reply by vdruts** [reply]
For me prompt weighting isn't working at all. :0 seems to just make it 100%

**Reply by uw11** [reply]
it works for me, i haven't updated the repo for a few days though
try it like this:
"prompt1 :0.9 prompt2: 0.1"
"prompt1 :0.5 prompt2: 0.5"
"prompt1 :0.1 prompt2: 0.9"
using the same seed you should be able to see that the image transforms between the two prompts

or you can use it in a different way like:
"cyberpunk: 20 very creative and detailed prompt to get what you want :40"
then you can adjust the 20 and 40 to push the generation to either direction
if you omit the last one, it will default to "1" weight iirc

:0 definitely disables the whole prompt-section that it refers to
![Screenshot_441](https://user-images.githubusercontent.com/810084/188877283-fc848a52-0f94-4c57-8fce-7e5734d75ad5.png)

**Comment by underlines** [comment]
I find this solution more intuitive: https://github.com/AUTOMATIC1111/stable-diffusion-webui-feature-showcase#attention

Using different type of brackets to increase/decrease weights of parts of the sentence relative to the others. Giving absolute numbers is a bit  confusing, because if it's not adding up to 1.0 or 100% it will be shifted anyways.

`A [Star wars] x-wing starfighter fighting against the (star trek) ((enterprise))
`
Where [..] is reducing and (..) increasing weights and can be combined.

**Reply by Gobz** [reply]
> I find this solution more intuitive: https://github.com/AUTOMATIC1111/stable-diffusion-webui-feature-showcase#attention

Definitely, there is also the question of adding negative prompts which seems to be a very useful feature.

**Comment by Tollanador** [comment]
It's important when using the weighted prompts to include a space before the colon, ie This is correct :0.25 vs  This is incorrect:0.75
